# Data Collection Notebook
This notebook contains the code that I used to collect the data for the different job titles. The job titles that I'll be webscraping are:
* [(1) Data Scientist](#point_1)
* [(2) Data Engineer](#point_2)
* [(3) Data Analyst](#point_3)
* [(4) Business Intelligence Analyst](#point_4)
* [(5) Quantitative Analyst](#point_5)
* [(6) Technology Consultant](#point_6)

In [3]:
#This is the glassdoorscraper notebook that we created in this repository as well
import glassdoor_scraper as gs
import pandas as pd

<a id='point_1'> </a>
## 1. Data Scientist
A quick search on glassdoor reveals that there are over 928 jobs on glassdoor relating to the keyword phrase Data Scientist. We pick the first 800 searches to improve the relevance of the search

In [ ]:
url1 = "https://www.glassdoor.sg/Job/singapore-data-scientist-jobs-SRCH_IL.0,9_IC3235921_KO10,24.htm"
df1 = gs.get_jobs(url1,800, False)

Progress: 0/800
Progress: 1/800
Progress: 2/800
Progress: 3/800
Progress: 4/800
Progress: 5/800
Progress: 6/800
Progress: 7/800
Progress: 8/800
Progress: 9/800
Progress: 10/800
Progress: 11/800
Progress: 12/800
Progress: 13/800
Progress: 14/800
Progress: 15/800
Progress: 16/800
Progress: 17/800
Progress: 18/800
Progress: 19/800
Progress: 20/800
Progress: 21/800
Progress: 22/800
Progress: 23/800
Progress: 24/800
Progress: 25/800
Progress: 26/800
Progress: 27/800
Progress: 28/800
Progress: 29/800
Progress: 30/800
Progress: 31/800
Progress: 32/800
Progress: 33/800
Progress: 34/800
Progress: 35/800
Progress: 36/800
Progress: 37/800
Progress: 38/800
Progress: 39/800
Progress: 40/800
Progress: 41/800
Progress: 42/800
Progress: 43/800
Progress: 44/800
Progress: 45/800
Progress: 46/800
Progress: 47/800
Progress: 48/800
Progress: 49/800
Progress: 50/800
Progress: 51/800
Progress: 52/800
Progress: 53/800
Progress: 54/800
Progress: 55/800
Progress: 56/800
Progress: 57/800
Progress: 58/800
Progres

Progress: 462/800
Progress: 463/800
Progress: 464/800
Progress: 465/800
Progress: 466/800
Progress: 467/800
Progress: 468/800
Progress: 469/800
Progress: 470/800
Progress: 471/800
Progress: 472/800
Progress: 473/800
Progress: 474/800
Progress: 475/800
Progress: 476/800
Progress: 477/800
Progress: 478/800
Progress: 479/800
Progress: 480/800
Progress: 481/800
Progress: 482/800
Progress: 483/800
Progress: 484/800
Progress: 485/800
Progress: 486/800
Progress: 487/800
Progress: 488/800
Progress: 489/800
Progress: 490/800
Progress: 491/800
Progress: 492/800
Progress: 493/800
Progress: 494/800
Progress: 495/800
Progress: 496/800
Progress: 497/800
Progress: 498/800
Progress: 499/800
Progress: 500/800
Progress: 501/800
Progress: 502/800
Progress: 503/800
Progress: 504/800
Progress: 505/800
Progress: 506/800
Progress: 507/800
Progress: 508/800
Progress: 509/800
Progress: 510/800
Progress: 511/800
Progress: 512/800
Progress: 513/800
Progress: 514/800
Progress: 515/800
Progress: 516/800
Progress: 

In [ ]:
df1.to_csv('Glassdoor Data Scientist Jobs.csv')

<a id='point_2'> </a>
## 2. Data Engineer
There were 4155 jobs listed on glassdoor relating to the keywords of **Data Engineer**, we'll take the first 2000 jobs.

In [ ]:
url2 = "https://www.glassdoor.sg/Job/singapore-data-engineer-jobs-SRCH_IL.0,9_IC3235921_KO10,23.htm"
df2 = gs.get_jobs(url2,20, False)

In [ ]:
df2.to_csv('Glassdoor Data Engineer Jobs.csv')

<a id='point_3'> </a>
## 3. Data Analyst
There were 246 jobs listed on glassdoor relating to the keywords of **Data Analyst**, we'll take the first 200 jobs.

In [ ]:
url3 = "https://www.glassdoor.sg/Job/singapore-data-analyst-jobs-SRCH_IL.0,9_IC3235921_KO10,22.htm"
df3 = gs.get_jobs(url,200, False)

In [ ]:
df3.to_csv('Glassdoor Data Analyst Jobs.csv')

<a id='point_4'> </a>
## 4. Business Intelligence Analyst
There were 1267 jobs listed on glassdoor relating to the keywords of **Business Intelligence Analyst**, we'll take the first 1000 jobs.

In [ ]:
url4 = "https://www.glassdoor.sg/Job/singapore-business-intelligence-analyst-jobs-SRCH_IL.0,9_IC3235921_KO10,39.htm"
df4 = gs.get_jobs(url,1000, False)

In [ ]:
df4.to_csv('Glassdoor Business Intelligence Analyst Jobs.csv')

<a id='point_5'> </a>
## 5. Quantitative Analyst
There were 491 jobs listed on glassdoor relating to the keywords of **Quantitative Analyst**, we'll take the first 400 jobs.

In [ ]:
url5 = "https://www.glassdoor.sg/Job/singapore-quantitative-analyst-jobs-SRCH_IL.0,9_IC3235921_KO10,30.htm"
df5 = gs.get_jobs(url,400, False)

In [ ]:
df5.to_csv('Glassdoor Quantitative Analyst Jobs.csv')

<a id='point_6'> </a>
## 6. Technology Consultant
There were 1820 jobs listed on glassdoor relating to the keywords of **Technology Consultant**, we'll take the first 1500 jobs.

In [ ]:
url6 = "https://www.glassdoor.sg/Job/singapore-technology-consultant-jobs-SRCH_IL.0,9_IC3235921_KO10,31.htm"
df6 = gs.get_jobs(url,1500, False)

In [ ]:
df6.to_csv('Glassdoor Quantitative Analyst Jobs.csv')